internal/docstore/mongodocstore/codec.go

```diff
@@ -49,17 +51,27 @@ type encoder struct {
 	val interface{}
 }
 
-func (e *encoder) EncodeNil()                                { e.val = nil }
-func (e *encoder) EncodeBool(x bool)                         { e.val = x }
-func (e *encoder) EncodeInt(x int64)                         { e.val = x }
-func (e *encoder) EncodeUint(x uint64)                       { e.val = int64(x) }
-func (e *encoder) EncodeBytes(x []byte)                      { e.val = x }
-func (e *encoder) EncodeFloat(x float64)                     { e.val = x }
-func (e *encoder) EncodeComplex(x complex128)                { e.val = x }
-func (e *encoder) EncodeString(x string)                     { e.val = x }
-func (e *encoder) ListIndex(int)                             { panic("impossible") }
-func (e *encoder) MapKey(string)                             { panic("impossible") }
-func (e *encoder) EncodeSpecial(reflect.Value) (bool, error) { return false, nil } // no special handling
+func (e *encoder) EncodeNil()                 { e.val = nil }
+func (e *encoder) EncodeBool(x bool)          { e.val = x }
+func (e *encoder) EncodeInt(x int64)          { e.val = x }
+func (e *encoder) EncodeUint(x uint64)        { e.val = int64(x) }
+func (e *encoder) EncodeBytes(x []byte)       { e.val = x }
+func (e *encoder) EncodeFloat(x float64)      { e.val = x }
+func (e *encoder) EncodeComplex(x complex128) { e.val = []float64{real(x), imag(x)} }
+func (e *encoder) EncodeString(x string)      { e.val = x }
+func (e *encoder) ListIndex(int)              { panic("impossible") }
+func (e *encoder) MapKey(string)              { panic("impossible") }
+
+var typeOfGoTime = reflect.TypeOf(time.Time{})
+
+func (e *encoder) EncodeSpecial(v reflect.Value) (bool, error) {
+	// Treat time "specially" as itself (otherwise its BinaryMarshal method will be called).
+	if v.Type() == typeOfGoTime {
+		e.val = v.Interface()
```

Comment:
Should we truncate it in the encoder to match the native encoder?

Reply:
Not necessary: the docstore encoder first creates a `map[string]interface{}`, then passes that to the native encoder. So it will get truncated there.